internal/ui/ui_glfw.go

```diff
@@ -1299,7 +1466,10 @@ func (u *userInterfaceImpl) setFullscreen(fullscreen bool) {
 	}
 
 	// glfw.PollEvents is necessary for macOS to enable (*glfw.Window).SetPos and SetSize (#2296).
-	glfw.PollEvents()
+	// This polling causes issues on Linux and Windows when rapidly toggling fullscreen, so we only run it under macOS.
+	if runtime.GOOS == "darwin" {
+		glfw.PollEvents()
+	}
```

Comment:
Interesting. We need to file this as an independent issue and might need to backport the fix to the stable branch if this is serious. I'll take a look at this later (or you can file the issue).